### ISSUE TITLE
Error preparing player items

### ISSUE THREAD

**Issue by lauw0129** [jarnedemeulemeester/findroid#10]
![Screenshot_2021-08-21-10-17-41-360_dev jdtech jellyfin](https://user-images.githubusercontent.com/62137079/130307508-7db46040-39e7-426b-bd69-777255e5425a.jpg)

This happened in some of the movie or TV shows. But some of those can play. I don't know what is the problem actually. Please help

**Comment by jarnedemeulemeester** [comment]
Hmm interesting...

Is there any way you could give me access to your Jellyfin instance? That way I can test it and figure out what the problem is.
If you can, just email me the url and login credentials. (jarnedemeulemeester@gmail.com)
Or Reddit dm whatever you prefer. (JDTechn0)

If not then no problem, but I really have no idea what the issue could be. 
Maybe there is a episode with no mediasource or something.
Either way, it's an issue with the app and I would love to fix it :)

**Comment by Derkades** [comment]
I have the same issue with a particular video, would it help if I sent the video file?

**Comment by lauw0129** [comment]
> Hmm interesting...
> 
> Is there any way you could give me access to your Jellyfin instance? That way I can test it and figure out what the problem is.
> If you can, just email me the url and login credentials. ([jarnedemeulemeester@gmail.com](mailto:jarnedemeulemeester@gmail.com))
> Or Reddit dm whatever you prefer. (JDTechn0)
> 
> If not then no problem, but I really have no idea what the issue could be.
> Maybe there is a episode with no mediasource or something.
> Either way, it's an issue with the app and I would love to fix it :)

I sent you the url and login credentials to you. Thanks for helping me.

**Comment by lauw0129** [comment]
Thanks for updating the app. But I just tried and it's also showing the same error message. Do you think it's because of it is H. 265?

**Comment by jarnedemeulemeester** [comment]
@lauw0129 Hmm on two devices I tested the video player opened (didn't play because of codec issue but that's a separate issue).
I tested with the first episode of Your Lie in April with a Pixel 1 & 3. 

@lauw0129 @Derkades In v0.1.1 there should be a new button next to the error message called "View details" (or "@string/view_details" because I made an oopsie). If you click on it a dialog should open with a detailed error message which you can share here :)

**Comment by lauw0129** [comment]
![Screenshot_2021-08-23-18-41-55-046_dev jdtech jellyfin](https://user-images.githubusercontent.com/62137079/130434527-dffa184a-55cf-44b0-81eb-0d028ce0c2bf.jpg)
But there's no any details at that button

**Comment by jarnedemeulemeester** [comment]
Looks like an IndexOutOfBoundsException. I will do some more testing and keep you updated.

**Comment by lauw0129** [comment]
Ok thank you

**Comment by jarnedemeulemeester** [comment]
@lauw0129 I am really struggling to find the cause of the problem.
I tested on many different devices, including an emulator with Android 7.0 (minimum supported by the app) and all worked.

What device are you using?
- Device name:
- Android version:

Can you also try using the user you provided me? To do this you have to remove your current server and add it again.
`Settings -> Manage servers -> Long press the server -> Remove`
Then add your server again but with the user 'jarne' you gave me. Maybe some configuration is different between the user you were using and the one you gave me.

I really want to solve this issue and I know on which line of code the error happens but I don't know why.

Let me know if my user works or not and what device you are using. 
I appreciate the effort and sorry for the trouble!

**Comment by lauw0129** [comment]
I just tried to login your account (jarne) in my server and it's work. But I don't know why it's not working in my account. I am checking the config. By the way thanks for your app. It's cool.

**Comment by lauw0129** [comment]
Could you see the video when you tried?

**Comment by lauw0129** [comment]
https://user-images.githubusercontent.com/62137079/130472635-a9d74a6c-ee38-477f-8ec0-7f1ad47af06d.mp4

It's from my friend's

**Comment by jarnedemeulemeester** [comment]
Yes I see. When I try on my devices I also just see a black screen.
It's probably because the media uses H.264 High 10 profile which can't be played on some devices.
This is something I can't fix immediately, I would have to implement a different player like libVLC or libMPV.
This is also a separate issue.

Now back to the "Error preparing player items" error.
Could you please tell me what is different between the two accounts? Maybe some setting?
It's strange that my account doesn't show this error but yours does.

**Comment by lauw0129** [comment]
There don't have any error in the new account. It just showed black screen. It could be the the media problem. But can it be solved by transcoding? I think transcode in the Findroid app may help to solve this problem

**Comment by Derkades** [comment]
Do you know where the indexoutofbounds error occurs specifically (which array lookup)? Does it print the stacktrace to a log somewhere? I've never tried accessing android logs but if it's useful to you I'll look into it.

**Comment by lauw0129** [comment]
I am not good at programming or something like those. So I also can't find where is the log.

**Comment by jarnedemeulemeester** [comment]
@Derkades This line:
https://github.com/jarnedemeulemeester/findroid/blob/e694f63b54bd71f241a44a205bd0cf84eb0de19c/app/src/main/java/dev/jdtech/jellyfin/viewmodels/EpisodeBottomSheetViewModel.kt#L84

It gets the first mediaSource from the mediaItem and it does this for all next episodes in the season. This to make a playlist for ExoPlayer to play all episodes after each other. This code could be much improved, and it will be!

Now I think I know why the problem occurs, @Derkades  do you have missing episodes enabled?
Episodes like this in the web ui: 
![image](https://user-images.githubusercontent.com/32322857/130691630-89338c3c-b2df-4740-a577-1501c1a134dc.png)

If this is the case then I will be fixing this tomorrow :)

**Comment by Derkades** [comment]
I do have missing episodes enabled but this specific episode only has one entry and it is not missing. Some other episodes in the same season do have duplicates (one of which is missing)

**Comment by jarnedemeulemeester** [comment]
@Derkades could you send me some screenshots of the season with the episodes in the web interface? I don't completely understand what you mean sorry

**Comment by Derkades** [comment]
Sure thing:
![image](https://user-images.githubusercontent.com/15892014/130698247-81760551-9429-4476-a47c-b1819b060231.png)

**Comment by jarnedemeulemeester** [comment]
Hmm okay I see. Any missing episodes after 14?

**Comment by Derkades** [comment]
No, episodes 15-18 all present

**Comment by jarnedemeulemeester** [comment]
I have added a fix for missing episodes. If there is a missing episode, it will just skip that one.
This should be in the next update.

But I'm not sure if that will have fixed your issue @Derkades 
Any way you can give me temporary access to your Jellyfin server to test it out? But I would understand if you wouldn't want to do that.

**Comment by Derkades** [comment]
Yes, it has been fixed. Thanks!
